src/Hl7.Fhir.Base/FhirPath/EvaluationContext.cs

```diff
@@ -29,6 +29,11 @@ public EvaluationContext(ITypedElement resource, ITypedElement rootResource)
             Resource = resource;
             RootResource = rootResource ?? resource;
         }
+        
+        public EvaluationContext(ITypedElement resource, ITypedElement rootResource, IDictionary<string, IEnumerable<ITypedElement>> environment) : this(resource, rootResource)
```

Comment:
I think we should also add this new parameter (optionally) to the extension methods (like POCO.Predicate, ITypedElement.Select etcetera).